apps/examples/webserver_test/webserver_test_main.c

```diff
@@ -0,0 +1,1323 @@
+
+errout:
+	if (param->callback && param->response) {
+		ws_test_http_client_response_release(param->response);
+	}
+	if (param->tls) {
+		ws_test_tls_ssl_release(client_tls);
+	}
+errout_before_tlsinit:
+	if (param->tls) {
+		ws_test_tls_release(client_tls);
+	}
+
+	free(client_tls);
+	if (!param->tls && sockfd > 0) {
+		close(sockfd);
+	}
+	param->async_flag = WS_TEST_ERR;
+	FUNC_EX;
+	return WS_TEST_ERR;
+}
+#endif
+
+static int ws_test_http_client_send_request(struct http_client_request_t *request,
+		 void *ssl_config, struct http_client_response_t *response, wget_callback_t cb)
+{
+	FUNC_EN;
+	int res = WS_TEST_ERR;
+#ifdef CONFIG_NET_SECURITY_TLS
+	struct mallinfo data;
+	struct http_client_ssl_config_t *ssl_conf = ssl_config;
+#endif
+
+	if (request == NULL) {
+		PRNT("Error: Request is null");
+		goto ret_error;
+	}
+
+	if (request->url == NULL) {
+		PRNT("Error: URL is NULL!!");
+		goto ret_error;
+	}
+
+	if (request->method < WS_TEST_MODE_GET || request->method > WS_TEST_MODE_DELETE) {
+		PRNT("Error: Incorrect method value!!");
+		goto ret_error;
+	}
+
+	if (request->encoding != CONTENT_LENGTH && request->encoding != CHUNKED_ENCODING) {
+		PRNT("Error: Incorrect encoding value!!");
+		goto ret_error;
+	}
+
+	request->buffer = (char *)zalloc(request->buflen);
+	if (request->buffer == NULL) {
+		PRNT("Error: Fail to allocate buffer");
+		goto ret_error;
+	}
+
+	if (request->buflen <= 0) {
+		PRNT("Error: Buffer length must be bigger than 0!!");
+		return WS_TEST_ERR;
+	}
+
+	request->tls = false;
+	request->response = response;
+	request->callback = cb;
+
+#ifdef CONFIG_NET_SECURITY_TLS
+	if (ssl_conf) {
+		request->tls = true;
+		memcpy(&request->ssl_config, ssl_conf, sizeof(struct http_client_ssl_config_t));
+
+		data = mallinfo();
+		if (data.fordblks < WS_TEST_CONF_MIN_TLS_MEMORY) {
+			PRNT("Error: Not enough memory!!");
+			goto ret_error;
+		}
+	}
+#endif
+
+	res = ws_test_base((void *)request);
+	if (res != 0) {
+		PRNT("ws_test_base() failed %d", res);
+	}
+
+	free(request->buffer);
+	FUNC_EX;
+	return res;
+
+ret_error:
+	FUNC_EX;
+	return WS_TEST_ERR;
+}
+
+static int ws_test_prepare_request(int argc, char** argv, struct http_client_request_t *request)
+{
+	FUNC_EN;
+	char *p, *q;
+	g_is_test_entity = 0;
+	g_keep_alive_count = 0;
+	g_chunk_count = 0;
+	g_trailer = 0;
+	memset(request, 0, sizeof(struct http_client_request_t));
+
+
+	if (!strncmp(argv[2], "GET", 4)) {
+		request->method = WS_TEST_MODE_GET;
+	} else if (!strncmp(argv[2], "PUT", 4)) {
+		request->method = WS_TEST_MODE_PUT;
+	} else if (!strncmp(argv[2], "POST", 5)) {
+		request->method = WS_TEST_MODE_POST;
+	} else if (!strncmp(argv[2], "DELETE", 7)) {
+		request->method = WS_TEST_MODE_DELETE;
+	} else {
+		PRNT("invalid req_type");
+		return WS_TEST_ERR;
+	}
+
+	request->url = argv[3];
+
+#ifdef CONFIG_NET_SECURITY_TLS
+	if (!strncmp(request->url, "https", 5)) {
+		g_https = 1;
+	} else
+#endif
+	if (!strncmp(request->url, "http", 4)) {
+		g_https = 0;
```

Comment:
please minimize global variable and add g_https in structure http_client_request_t

Reply:
done